### PR TITLE
Feat/add row edit modal

### PR DESCRIPTION
**Pull request by MaxPostema**

### CODE REVIEW

File: apps/molgenis-components/src/client/client.js [hunk 1]
```diff
@@ -154,35 +207,64 @@ const request = async (url, graqphl) => {
...
+  return new Promise(function (resolve, reject) {
+    //thanks to https://medium.com/@danielbuechele/file-uploads-with-graphql-and-apollo-5502bbf3941e
+    const formData = new FormData();
+
+    // search for File objects on the request and set it as formData
+    for (let { node, path } of new RecursiveIterator(variables)) {
+      if (node instanceof File) {
+        const id = Math.random().toString(36);
+        formData.append(id, node);
+        objectPath.set(variables, path.join("."), id);
```
Comment (connoratrug) [inline apps/molgenis-components/src/client/client.js:235]: is this worth adding a decency, the objectPath seems to have a single core contributor and a (high) security alert . Could be solve this without the lib ?

File: apps/molgenis-components/src/client/client.js [hunk 2]
```diff
@@ -154,35 +207,64 @@ const request = async (url, graqphl) => {
...
+  return `mutation ${action}($value:[${name}Input]){${action}(${name}:$value){message}}`;
+}
+
+function requestMultipart(url, query, variables) {
+  return new Promise(function (resolve, reject) {
+    //thanks to https://medium.com/@danielbuechele/file-uploads-with-graphql-and-apollo-5502bbf3941e
+    const formData = new FormData();
+
+    // search for File objects on the request and set it as formData
+    for (let { node, path } of new RecursiveIterator(variables)) {
```
Comment (connoratrug) [inline apps/molgenis-components/src/client/client.js:231]: is this worth adding a dependency ? ( looks like a copy from the medium post)

File: apps/molgenis-components/src/components/forms/EditModal.vue [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+  },
+  props: {
+    id: {
+      type: String,
+      required: true,
+    },
+    pkey: Object,
+    clone: Boolean,
+    visibleColumns: Array,
+    defaultValue: Object,
```
Comment (connoratrug) [inline apps/molgenis-components/src/components/forms/EditModal.vue:80]: required ? default ?

File: apps/molgenis-components/src/components/forms/EditModal.vue [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+    ButtonOutline,
+  },
+  methods: {
+    saveDraft() {
+      this.save(true);
+    },
+    async save(isDraft) {
+      let promise;
+      if (this.pkey && !this.clone) {
+        promise = this.client.updateVariables(
```
Comment (connoratrug) [inline apps/molgenis-components/src/components/forms/EditModal.vue:107]: maybe do `
const saveMethod = this.pkey && !this.clone ? 'updateVariables' : 'insertVariable'
await this.client[saveMethod](params)
`

or some type of apply ?

File: apps/molgenis-components/src/client/client.js (hunk 1 above)
Comment (chinook25) [reply apps/molgenis-components/src/client/client.js:235]: No clue, just did copy paste to get stuff working. Something to look into when we'll tidy up the code.

File: apps/molgenis-components/src/client/client.js (hunk 2 above)
Comment (chinook25) [reply apps/molgenis-components/src/client/client.js:231]: Same answer ;D

File: apps/molgenis-components/src/components/forms/EditModal.vue (hunk 3 above)
Comment (chinook25) [reply apps/molgenis-components/src/components/forms/EditModal.vue:80]: Copy paste, haven't gotten there yet.

**Pull request closed without merging**